lua-functions.cc

```diff
@@ -20,12 +20,12 @@ int listDirectoryFull(lua_State *L, int mode) {
 	WIN32_FIND_DATAA fdFile;
 	HANDLE handle = NULL;
 
-	std::string path = format("%s\\*.*", dirname);
+	std::string path = format(".\\%s\\*.*", dirname);
```

Comment:
Neat, though what I had in mind was more along the lines of preventing mods from accessing anything outside of the game's directory (and maybe saves directory) altogether. I guess we'd need to resolve the path and check it against a whitelist, or something

Reply:
I see. This change will force the path to always start at the game's directory. However, it is possible to use the path `..` to go back one step (or several steps if used multiple times). For example `os.listdirs("../Among Us")` would list the directories in that game instead.

Reply:
Could disallow usage of `..` with the following if you want:
```lua
if ((int)path.find("..") != -1 ) {
	return 1;
}
```

Reply:
While outlawing `..` works, it might prevent some legitimate use cases. Let's drop this for now - even if we successfully roll a whitelist, it can be trivially circumvented as long as the mod loader doesn't prevent access to `os.execute` and other such functions...